ui/app/mirrors/create/cdc/columnbox.tsx

```diff
@@ -20,69 +21,150 @@ export default function ColumnBox({
+    const rowIndex = rows.findIndex((row) => row.source === tableRow.source);
     if (rowIndex !== -1) {
-      const sourceRow = currRows[rowIndex],
-        newExclude = new Set(sourceRow.exclude);
+      const updatedRows = [...rows];
+      updatedRows[rowIndex] = updater(updatedRows[rowIndex]);
+      setRows(updatedRows);
+    }
+  };
+
+  const handleColumnExclusion = (column: string, include: boolean) => {
+    updateRow((row) => {
+      const newExclude = new Set(row.exclude);
       if (include) {
         newExclude.delete(column);
       } else {
         newExclude.add(column);
       }
-      currRows[rowIndex] = {
-        ...sourceRow,
-        exclude: newExclude,
-      };
-      setRows(currRows);
-    }
+      return { ...row, exclude: newExclude };
+    });
   };
 
-  return columns.map((column) => {
-    const partOfOrderingKey = rows
-      .find((row) => row.source == tableRow.source)
-      ?.columns.some(
-        (col) =>
-          col.sourceName === column.name &&
-          (col.ordering > 0 || col.partitioning > 0)
+  const handleNullableEnabledChange = (columnName: string, enabled: boolean) => {
+    updateRow((row) => {
+      const existingColumn = row.columns.find(
+        (col) => col.sourceName === columnName
       );
+
+      const updatedColumns: ColumnSetting[] = existingColumn
+        ? // Update existing ColumnSetting
+        row.columns.map((col) =>
+          col.sourceName === columnName
+            ? { ...col, nullableEnabled: enabled }
+            : col
+        )
+        : // Create new ColumnSetting
+        [
+          ...row.columns,
+          {
+            sourceName: columnName,
+            destinationName: '',
+            destinationType: '',
+            ordering: 0,
+            partitioning: 0,
+            nullableEnabled: enabled,
+          },
+        ];
+
+      return { ...row, columns: updatedColumns };
+    });
+  };
+
+  const getNullableEnabled = (columnName: string): boolean => {
     return (
-      <RowWithCheckbox
-        key={column.name}
-        label={
-          <Label
-            as='label'
-            style={{
-              fontSize: 13,
-              display: 'flex',
-              alignItems: 'center',
-            }}
-          >
-            {column.name}
-            <p
+      tableRow.columns.find((col) => col.sourceName === columnName)
+        ?.nullableEnabled ?? false
+    );
+  };
+
+  return (
+    <div
+      style={{
+        display: 'grid',
+        gridTemplateColumns: showOrdering
+          ? 'minmax(0, 2fr) minmax(0, 1fr) auto'
+          : 'minmax(0, 2fr) minmax(0, 1fr)',
+        alignItems: 'center',
+        columnGap: '10rem',
+        width: '80%',
+      }}
+    >
+      <div style={{ fontSize: 12, fontWeight: 500 }}>Name</div>
+      <div style={{ fontSize: 12, fontWeight: 500 }}>Type</div>
+      {showOrdering &&
+        <div style={{ fontSize: 12, fontWeight: 500 }}>Nullable</div>
```

Comment:
```suggestion
        <div style={{ fontSize: 12, fontWeight: 500, display: 'flex', alignItems: 'center', gap: '0.25rem' }}>
          Nullable
          <InfoPopover tips="Enabling nullable columns wraps types in Nullable() which may impact query performance and storage in ClickHouse." />
        </div>
```

Reply:
Sorry, didn't realize the popover would land right on top of checkboxes and would make things look off. Not seeing a way to make it all work together, let's just remove it.